src/BloomTests/web/EnhancedImageServerTests.cs

```diff
@@ -173,6 +173,65 @@ public void CanRetrieveContentOfFakeTempFile_ButOnlyUntilDisposed()
 			}
 		}
 
+		[Test]
+		public void CanRetrieveContentOfFakeTempFile_WhenFolderContainsAmpersand_ViaJavaScript()
+		{
+			const bool simulateCallingFromJavaScript = true;
+			using (var server = CreateImageServer())
+			{
+				var ampSubfolder = Path.Combine(_folder.Path, "Using &lt;, &gt;, & &amp; in HTML");
+				Directory.CreateDirectory(ampSubfolder);
+				var html = @"<html ><head><title>Using &lt;lt;, &gt;gt;, &amp; &amp;amp; in HTML</title></head><body>here it is</body></html>";
+				var dom = new HtmlDom(html);
+				dom.BaseForRelativePaths = ampSubfolder.ToLocalhost();
+				string url;
+				using (var fakeTempFile = EnhancedImageServer.MakeSimulatedPageFileInBookFolder(dom, simulateCallingFromJavaScript))
+				{
+					url = fakeTempFile.Key;
+					var transaction = new PretendRequestInfo(url, false, simulateCallingFromJavaScript);
+
+					// Execute
+					server.MakeReply(transaction);
+
+					// Verify
+					// Whitespace inserted by CreateHtml5StringFromXml seems to vary across versions and platforms.
+					// I would rather verify the actual output, but don't want this test to be fragile, and the
+					// main point is that we get a file with the DOM content.
+					Assert.That(transaction.ReplyContents,
+						Is.EqualTo(TempFileUtils.CreateHtml5StringFromXml(dom.RawDom)));
+				}
+			}
+		}
+
+		[Test]
+		public void CanRetrieveContentOfFakeTempFile_WhenFolderContainsAmpersand_NotViaJavaScript()
+		{
```

Comment:
These two tests still look almost identical...can we factor out the common code and just pass a simulateCallingFromJavaScript argument?